witness/golang/cmd/witness/internal/witness/witness.go

```diff
@@ -121,120 +121,113 @@ func (w *Witness) GetLogs() ([]string, error) {
 }
 
 // GetCheckpoint gets a checkpoint for a given log, which is consistent with all
-// other checkpoints for the same log signed by this witness.  It also signs it
-// under the witness' key.
+// other checkpoints for the same log signed by this witness.
 func (w *Witness) GetCheckpoint(logID string) ([]byte, error) {
 	chkpt, _, err := w.getLatestChkptData(w.db.QueryRow, logID)
 	if err != nil {
 		return nil, err
 	}
-	// Add the witness' signature to the checkpoint.
-	logInfo, ok := w.Logs[logID]
-	if !ok {
-		return nil, fmt.Errorf("log %q not found", logID)
-	}
-	n, err := note.Open(chkpt, note.VerifierList(logInfo.SigVs...))
-	if err != nil {
-		return nil, fmt.Errorf("failed to verify checkpoint: %v", err)
-	}
-	cosigned, err := note.Sign(n, w.Signer)
-	if err != nil {
-		return nil, fmt.Errorf("couldn't sign checkpoint: %v", err)
-	}
-	return cosigned, nil
+	return chkpt, nil
 }
 
 // Update updates the latest checkpoint if nextRaw is consistent with the current
-// latest one for this log. It returns the latest checkpoint size before the
-// update was applied (or just what was fetched if the update was unsuccessful).
-func (w *Witness) Update(ctx context.Context, logID string, nextRaw []byte, proof [][]byte) (uint64, error) {
+// latest one for this log. If the update was successful it returns a cosigned
+// version of nextRaw and if not it returns its (signed) latest checkpoint.
```

Comment:
```suggestion
// latest one for this log. This returns the witness's latest cosigned checkpoint,
// which will be a signed version of nextRaw if this update was applied.
```

I think the important property is that whatever it returns is whatever the next call to update should use as its base (unless the world changes again). I think my suggestion is a little clearer on that, but I'm not attached to it.

Reply:
Okay, that's a nice way of thinking about it.  Used your text mostly verbatim.